### PR TITLE
Mobt 157 nbhood refactor consolidate unit tests part1

### PR DESCRIPTION
**Pull request by fionaRust**
Part one of this PR https://github.com/metoppv/improver/pull/1664

### CODE REVIEW

**Comment by fionaRust** [comment]
One unit test needs updating following the rebase after the previous PR was merged. I'll fix this along with any other review comments after the testbed.

File: improver_tests/nbhood/nbhood/test_GeneratePercentilesFromANeighbourhood.py [hunk 1]
```diff
@@ -28,62 +28,725 @@
...
+        """Test behaviour with a masked non-zero point.
+        The underlying numpy.percentile base behaviour does not support
+        masked arrays so raise an error.
+        """
+        mask = np.zeros_like(self.cube.data)
+        mask[2, 2] = 1
+        self.cube.data = np.ma.masked_array(self.cube.data, mask=mask)
+        percentiles = np.array([10, 50, 90])
+        radius = 2000.0
+        message = "The use of masked data is not yet implemented."
```
Comment (gavinevans) [inline improver_tests/nbhood/nbhood/test_GeneratePercentilesFromANeighbourhood.py:487]: As noted on the PR, this message requires updating.

**Review by gavinevans — CHANGES_REQUESTED** [review]
Thanks @fionaRust 👍 

As a stepping stone towards the refactoring of the neighbourhood processing plugins, I think that this is fine. I've added a couple of very minor comments, one of which is already noted on the ticket.

For another reviewer: The size of the diff in this PR is much larger than necessary, as `improver_tests/nbhood/circular_kernel/test_GeneratePercentilesFromACircularNeighbourhood.py` and `improver_tests/nbhood/nbhood/test_GeneratePercentilesFromANeighbourhood.py` are almost identical.

**Review by Kat-90 — COMMENTED** [review]
Gone through the code and pulled to run the tests. As Gavin mentioned, there is one failing test related to test_GeneratePercentileFromANeighbourhood.py but all other tests passed. 

I just have 2 small comments.

File: improver_tests/nbhood/nbhood/test_GeneratePercentilesFromANeighbourhood.py (hunk 1 above)
Comment (fionaRust) [reply improver_tests/nbhood/nbhood/test_GeneratePercentilesFromANeighbourhood.py:487]: Done

File: improver_tests/precipitation_type/convection/test_DiagnoseConvectivePrecipitation.py [hunk 2]
```diff
@@ -213,6 +213,7 @@ def test_circular_neighbourhood(self):
             self.higher_threshold,
             neighbourhood_method,
             self.radii,
+            weighted_mode=True,
         )._calculate_convective_ratio(self.cubelist, self.threshold_list)
         self.assertArrayAlmostEqual(result, expected)
```
[the remark this answers is not in the document]
Comment (fionaRust) [reply improver_tests/precipitation_type/convection/test_DiagnoseConvectivePrecipitation.py:219]: I've updated the docstring in the test below, as you're right it wasn't correct. 

These tests are testing a method on the DiagnoseConvectivePrecipitation that uses neighbourhood processing, so these tests are testing something different to the test_NeighbourhoodProcessing. See:
https://github.com/metoppv/improver/blob/master/improver/precipitation_type/convection.py#L143

I'm also not too worried about these tests as the plugin is going to be removed in https://github.com/metoppv/mo-blue-team/issues/214

**Review by gavinevans — APPROVED** [review]

File: improver_tests/precipitation_type/convection/test_DiagnoseConvectivePrecipitation.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Kat-90) [reply improver_tests/precipitation_type/convection/test_DiagnoseConvectivePrecipitation.py:219]: Thanks for explaining and thanks for the updated doc string.

File: improver/nbhood/square_kernel.py [hunk 3]
```diff
@@ -101,7 +101,12 @@ def __init__(
             msg = "{} is not a valid neighbourhood_method.".format(neighbourhood_method)
             raise ValueError(msg)
         if weighted_mode and neighbourhood_method != "circular":
-            msg = "weighted_mode can only be used if neighbourhood_method is circular"
+            msg = (
+                "weighted_mode can only be used if neighbourhood_method is circular."
+                f" weighted_mode provided: {weighted_mode}, "
+                f"neighbourhood_method provided: {neighbourhood_method}."
+            )
+            print(msg)
```
Comment (Kat-90) [inline improver/nbhood/square_kernel.py:109]: Think this print needs removing.

File: improver/nbhood/square_kernel.py [hunk 4]
```diff
@@ -105,23 +100,19 @@ def __init__(
...
-                "The neighbourhood output can either be in the form of a "
-                "sum of all the points in the neighbourhood or a fraction "
-                "of the sum of the neighbourhood divided by the "
-                "neighbourhood area. The {} option is invalid. "
-                "Valid options are 'sum' or 'fraction'."
+                "weighted_mode can only be used if neighbourhood_method is circular."
+                f" weighted_mode provided: {weighted_mode}, "
+                f"neighbourhood_method provided: {neighbourhood_method}."
             )
+            print(msg)
```
Comment (fionaRust) [inline improver/nbhood/square_kernel.py:109]:
```suggestion
```

**Review by Kat-90 — APPROVED** [review]
Happy to approve

**Pull request merged**